advtraj/integrate/__init__.py

```diff
@@ -51,39 +51,27 @@ def _validate_starting_points(ds):
             f" {', '.join(missing_vars)}"
         )
 
+    if "time" not in ds.coords:
+        raise Exception("The starting position dataset is missing the time cooord.")
 
-def _integrate_single_trajectory(ds_starting_point, ds_position_scalars, da_times):
-    """
-    Integrate a single trajectory both forward and backward from
-    `ds_starting_point` usnig `ds_position_scalars` and times in `da_times`
-    """
-    da_times_backward = da_times.sel(time=slice(None, ds_starting_point.time))
-    da_times_forward = da_times.sel(time=slice(ds_starting_point.time, None)).isel(
-        time=slice(1, None)
-    )
+    coords = [c for c in ds[POSITION_COORD_NAMES[0]].coords if "time" not in c]
```

Comment:
It's not clear to me what this line is doing, sorry. Is it checking the coordinates of the `x` coordinate of the supplied dataset? Does that even make sense? :)

I don't think this is needed since an exception will be raised if any of the required coordinates a missing. Or do we need to have an extra check here?

Reply:
As the raised exception suggests its checking that the starting points data_vars ("x", "y", "z") have a coordinate in addition to time; note a coordinate, not a dimension. This should be called 'trajectory_number', and we could make the code stricter, but actually the default 'dim_0' will do. Unfortunately, if you only have a single trajectory, you have to force xarray to include a coord for it, so this is checking its there. We could just make this a test for 'trajectory_number' if you like.
Incidentally, while 'trajectory_number' is clear, its a bit long. Any preferences for a shorter name?